proposals/redirects.md

```diff
@@ -0,0 +1,116 @@
+<!--
+  Note: You are probably looking for `stage-1--discussion-template.md`!
+  This template is reserved for anyone championing an already-approved proposal.
+
+  Community members who would like to propose an idea or feature should begin
+  by creating a GitHub Discussion. See the repo README.md for more info.
+
+  To use this template: create a new, empty file in the repo under `proposals/${ID}.md`.
+  Replace `${ID}` with the official accepted proposal ID, found in the GitHub Issue
+  of the accepted proposal.
+-->
+
+- Start Date: 2023-05-22
+- Reference Issues: https://github.com/withastro/roadmap/issues/466
+- Implementation PR: https://github.com/withastro/astro/pull/7067/
+
+# Summary
+
+Add a `redirects` config option to the Astro config which allows you to define redirects in a central place. Allow integrations to read this information and apply it to it's own redirects configuration for deployment.
+
+# Example
+
+New config option used like so:
+
+```js
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  redirects: {
+    '/other': '/place'
+  }
+});
```

Comment:
Is there a way to redirect multiple pages at once? e.g. `/blog/*` -> `/articles/*`.

Also, platforms like Cloudflare supports [advanced types of redirects](https://developers.cloudflare.com/pages/platform/redirects/), and if they have to resort to using a custom `_redirects` file for it (which presumably you can't use both the file and `redirects` config at the same time), it puts the `redirects` option in an odd spot.

Reply:
The intent is to support our file router's syntax, so you would do:

```js
'blog/[...params]': '/articles/[...params]'
```

I need to implement this before I write it into the RFC as there could be more to it.

Reply:
That sounds like a neat idea! I'll resolve this first then.